GMS/FullGMS.cpp

```diff
@@ -0,0 +1,193 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * Copyright 2022 (c) Alen Galinec
+ */
+
+#include <sstream>
+#include <string>
+#include <unordered_map>
+
+#include "../TypeDefinition/GMS/Constants.hpp"
+#include "../TypeDefinition/GMS/GMSType.hpp"
+#include "../TypeDefinition/TypeDefinition.hpp"
+#include "FullGMS.hpp"
+
+namespace {
+static std::unordered_map<std::uint32_t, UmatiServerLib::LocalizedText_t> CLASS_MAP = {
+  {0, {"en", "NoTool"}},
+  {1, {"en", "UnDefTool"}},
+  {2, {"en", "TactileTouchTrigger"}},
+  {3, {"en", "TactileMeasuring"}},
+  {4, {"en", "Optical-1D"}},
+  {5, {"en", "Optical-2D"}},
+  {6, {"en", "Optical-3D"}},
+  {7, {"en", "Roughness"}},
+  {8, {"en", "Eddy Current Sensor"}},
+  {9, {"en", "TemperatureProbing"}},
+  {10, {"en", "PtMeas"}},
+  {11, {"en", "Other"}}};
+}
+
+FullGMS::FullGMS(UA_Server *pServer) : InstantiatedGMS(pServer) {
+  MachineName = "FullGMS";
+  CreateObject();
+}
+
+void FullGMS::CreateObject() {
+  std::stringstream ss;
+  ss << "http://www.isw.uni-stuttgart.de/" << MachineName << "/";
+  m_nsIndex = UA_Server_addNamespace(m_pServer, ss.str().c_str());
+  UA_ObjectAttributes objAttr = UA_ObjectAttributes_default;
+
+  objAttr.displayName = UA_LOCALIZEDTEXT_ALLOC("", MachineName.c_str());
+  objAttr.eventNotifier = UA_EVENTNOTIFIERTYPE_SUBSCRIBETOEVENTS;
+
+  auto status = UA_Server_addObjectNode(
+    m_pServer,
+    UA_NODEID_NUMERIC(m_nsIndex, 0),
+    UA_NODEID_NUMERIC(nsFromUri(m_pServer, constants::NsMachineryUri), UA_MACHINERYID_MACHINES),
+    UA_NODEID_NUMERIC(0, UA_NS0ID_ORGANIZES),
+    *open62541Cpp::UA_QualifiedName(m_nsIndex, MachineName).QualifiedName,
+    UA_NODEID_NUMERIC(nsFromUri(m_pServer, constants::NsGMSUri), UA_GMSID_GMSTYPE),
+    objAttr,
+    nullptr,
+    m_nodeId.NodeId);
+
+  UA_ObjectAttributes_clear(&objAttr);
+  UmatiServerLib::Bind::MembersRefl(gms, m_pServer, m_nodeId, n);
+
+  InstantiateIdentification();
+  InstantiateMonitoring();
+  InstantiateProduction();
+  InstantiateResultManagement();
+  InstantiateEquipment();
+  InstantiateNotification();
+}
+
+void FullGMS::InstantiateIdentification() {
+  {
+    std::stringstream ss;
```

Comment:
Redudant to base class -> you better call here the InstantiateIdentification method of InstantiatedGMS class

Reply:
True! I should just change the values of already instantiated nodes.